app/code/Magento/InventorySales/Model/IsProductSalableForRequestedQtyCondition/BackOrderCondition.php

```diff
@@ -0,0 +1,34 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\InventorySales\Model\IsProductSalableForRequestedQtyCondition;
+
+use Magento\InventorySalesApi\Api\IsProductSalableForRequestedQtyInterface;
+
+/**
+ * @inheritdoc
+ */
+class BackOrderCondition implements IsProductSalableForRequestedQtyInterface
+{
+    /** @var \Magento\InventorySales\Model\IsProductSalableCondition\BackOrderCondition */
+    private $backOrderCondition;
+
+    public function __construct(
+        \Magento\InventorySales\Model\IsProductSalableCondition\BackOrderCondition $backOrderCondition
+    ) {
+        $this->backOrderCondition = $backOrderCondition;
+    }
+
+    /**
+     * @inheritdoc
+     * @SuppressWarnings(PHPMD.UnusedFormalParameter)
+     */
+    public function execute(string $sku, int $stockId, float $requestedQty): bool
```

Comment:
Looks like we need to return result object, not just typical bool
The same as for Validators.

Because we somehow need to return validation message, what exactly went wrong and what should be fixed, like in existing code:
```
        if ($stockItem->getMaxSaleQty() && $qty > $stockItem->getMaxSaleQty()) {
            $result->setHasError(true)
                ->setMessage(__('The most you may purchase is %1.', $stockItem->getMaxSaleQty() * 1))
                ->setErrorCode('qty_max')
                ->setQuoteMessage(__('Please correct the quantity for some products.'))
                ->setQuoteMessageIndex('qty');
            return $result;
        }
```

I propose to return next structure
```php
interface IsProductSalableResultInterface
{

    /**
     * @return bool
     */
    public function isSalable(): bool;

    /**
     * @return ProductSalabilityError[]
     */
    public function getErrors(): array;
}

interface ProductSalabilityError
{

    /**
     * @return string
     */
    public function getCode(): string;

    /**
     * @return string
     */
    public function getMessage(): string;
}
```
ProductSalabilityError entities are created with ProductSalabilityErrorFactory